Automation Examples/010.md

```diff
@@ -0,0 +1,200 @@
+
+#### 5. After you have opened a company webpage, wait for the elements to be loaded and click the first .pdf file in the table to download it. 
+
+After a quick investigation, you should see that opening the first .pdf file on this webpage is possible by running this script:
+
+```js
+document.getElementsByClassName("date")[2].getElementsByTagName("a")[0].click()
+```
+
+But obviously you remember about waiting for these class elements to be loaded so first you add this line of code:
+
+```G1ANT
+        selenium.waitforvalue ‴return document.getElementsByClassName("date").length > 0‴ expectedvalue true
+```
+
+And then you open the .pdf file like this:
+
+```G1ANT
+        selenium.runscript ‴document.getElementsByClassName("date")[2].getElementsByTagName("a")[0].click()‴
+```
+
+#### 6. The .pdf file has been opened in the second tab so save it under the correct folder (see the process description)
+
+```G1ANT
+        ♥dateformat = yyyy-MM-dd
+        ♥path = ♥basePath\♥date\♥companyName
+        keyboard ⋘CTRL+S⋙
+        ⊂
+        System.IO.Directory.CreateDirectory(♥path);
+        ⊃
+        keyboard ⋘CTRL+C⋙
+        ♥clipboard = ♥path\♥clipboard
+        keyboard ⋘CTRL+V⋙
+        keyboard ⋘ENTER⋙
+```
+
+The `dateformat` variable is a special variable which specifies the format of the today date which is stored by the `date` special variable. As we want to process 50 companies, it is nice to have this script execute each step as fast as it can so we use clipboard for that instead of typing everything key by key. To save this file, we use Ctrl+S shortcut and we create a directory using a C# snippet for this file. Using Ctrl+C shortcut we copy the name of the .pdf file (it is already selected), add the full path to the .pdf file to the `clipboard` variable, paste it using Ctrl+V shortcut and press Enter. 
+
+#### 7. Close the tab, go back to the main webpage and close the for block
+
+```G1ANT
+        selenium.closetab
+        selenium.seturl ♥url
+    end for 
+```
+
+#### 8. Close the selenium in between the `end for` command and `end procedure` command
+
+```G1ANT
+    end for
+    
+    selenium.close
+end procedure
+```
+
+#### 9. After running this script multiple times you notice the first issue - sometimes "Confirm Save As" window appears when you download the same file again. Fix this
+
+After pressing Enter add the following workaround.
+
+```G1ANT
+        try errorcall Empty
+            window ‴✱Confirm Save As✱‴
+            keyboard ⋘LEFT⋙⋘ENTER⋙
+        end try
+```
+
+If the window is displayed, click "yes" to save the file. If the window is not displayed, the script would fail so you surround this code with `try` block, add errorcall argument to it so that it runs smoothly without showing any error messages. Add the "Empty" procedure at the end of your script of course.
```

Comment:
```suggestion
If the window is displayed, click "yes" to save the file. If the window is not displayed, the script would fail so you surround this code with `try` block, add errorcall argument to it so that it runs smoothly without showing any error messages. Add the "Empty" procedure at the end of your script.
```